GoogleForCreators/web-stories-wp
# Quick Action Menus: Reset element

## Context

Adding more robust "Clear" or "Reset" to quick action menus feature.

## Summary

There was a bit of confusion about what all "Clear" was supposed to do - thinking it was just animations and filters given the design spec! This adds border, border radius and opacity to what is cleared/clearable on an element.  So the full list of "Styles" applied to an element that will show the "clear" button on a quick menu are:  **animation, overlay, border radius, border, and opacity** 

At present, opted to not clear/reset anything else because everything else is already present on an element when it's dropped into the canvas. Doesn't feel right to hijack the elements further, this is currently just clearing out the optional additions to an element which feels straight forward. I have a question out to the design team to confirm this. We can always add in more but it starts to get a little weird because some elements differ in their defaults. 

## Relevant Technical Choices

- Consolidate `clearAnimations` and `clearAnimationsAndFilters` to be `resetElement` 
- Fixed a prop type i goofed on 
- Add a specific check for text element border radius as this has a different default than other elements (good example of the added complexity on a very basic level).

## To-do

N/A

## User-facing changes

None, behind feature flag

## Testing Instructions

Play with quick action menus, apply some or all (go crazy!) of the things I mentioned above (animation, overlay, border radius, border, and opacity) and see the "clear' button show up on the selected element's quick action menu. Confirm that clear and undo work as expected. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #7707
